src/WebInterface.hs

```diff
@@ -173,6 +183,59 @@ viewProjectQueues info state = do
     h2 "Recently integrated"
     viewList viewPullRequestWithApproval info state integrated
 
+-- Render the html for the queues in a project, excluding the header and footer.
+viewGroupedProjectQueues :: [(ProjectInfo, ProjectState)] -> Html
+viewGroupedProjectQueues projects = do
+  let
+    pullRequests :: [((ProjectInfo, ProjectState), [(PullRequestId, Project.PullRequestStatus)])]
```

Comment:
This signature looked scary to me at first, but then I spent some time thinking about how it could be simplified, and now it actually makes a lot of sense to me. It is probably the fault of `viewList` that requires passing around the state everywhere. Maybe we could have and extended version of `classifyPullRequests` that also returns the pull request itself, then there would be no need for `viewList` to do the lookup, and many things get simpler.

If you like you can do that here, but merging this as-is is also fine with me, then I’ll look into it later.

Reply:
I simplified the types (and in the process removing a TODO for the `fromJust`), so now `viewList` doesn't require a state anymore.